libs/community/langchain_community/vectorstores/lancedb.py

```diff
@@ -96,6 +124,52 @@ def __init__(
                             "api key provided with local uri.\
                             The data will be stored locally"
                         )
+        if table is not None:
+            try:
+                assert isinstance(
+                    table, (lancedb.db.LanceTable, lancedb.remote.table.RemoteTable)
+                )
+                self._table = table
+                self._table_name = (
+                    table.name if hasattr(table, "name") else "remote_table"
+                )
+            except AssertionError:
+                raise ValueError(
+                    """`table` has to be a lancedb.db.LanceTable or 
+                    lancedb.remote.table.RemoteTable object."""
+                )
+        else:
+            self._table = self.get_table(table_name, set_default=True)
+
+    def results_to_docs(self, results: Any, score: bool = False) -> Any:
+        columns = results.schema.names
+
+        if "_distance" in columns:
+            score_col = "_distance"
+        elif "_relevance_score" in columns:
+            score_col = "_relevance_score"
+        else:
+            score_col = None
+
+        if score_col is None or not score:
+            return [
+                Document(
+                    page_content=results[self._text_key][idx].as_py(),
+                    metadata=results["metadata"][idx].as_py(),
```

Comment:
Should this be results[idx][self._text_key/"metadata"] instead?

Reply:
Same comment for 166-167, I might just be missing something

Reply:
Hi, its the arrow convention of slicing,  the results are a `pyarrow.Table` thats why `col_name` needs to precede idx